### ISSUE TITLE
Should libmodsec pass action clear m_actions?

### ISSUE THREAD

**Issue by p0pr0ck5** [SpiderLabs/ModSecurity#1152]
Examine the following test rules:

```
SecRule ARGS:foo "bar" "id:'900017',phase:1,t:none,deny,nolog,msg:'foo = bar'"
SecRule &TX:REAL_IP "@eq 0" "id:'900021',phase:1,t:none,initcol:global=global,initcol:ip=%{remote_addr}_%{tx.ua_hash},setvar:tx.real_ip=%{remote_addr},nolog,pass
```

Suppose that both rules match. Calling `msc_intervention` against this after `msc_process_request_headers` will not result in the transaction being intervened, because `pass` clears `m_actions`. Obviously, removing `pass` from 900021 does result in `msc_intervention` returning true because `m_actions` hasn't been cleared. Is this intended behavior? _Should_ it be?

I would think that `pass` would essentially function like an 'ignore' where processing continues and nothing really happens, but here we see a tangible effect on the transaction.

**Comment by zimmerle** [comment]
Hi @p0pr0ck5,

Fixed by: 3d1d051. Want to double check?

There is a test case at: 
https://github.com/SpiderLabs/ModSecurity/blob/libmodsecurity/test/test-cases/regression/issue-1152.json

**Comment by p0pr0ck5** [comment]
Sure, will give it a look over early next week (holiday weekend for us :) )
